### PR TITLE
Develop

### LINKED ISSUE #33: Product types can be duplicated in table
For property in the product table if revenue and non-revenue line is selected then we get 2 entries in the report table.

### LINKED ISSUE #37: Fix hardcoding of recatpcha code in UI
See tab_auth_ui function

### PR DESCRIPTION
**Pull request by krzmip**
closes #33 

edit: also closes #37

### CODE REVIEW

**Comment by krzmip** [comment]
pull request updated, now it also allows passing arguments to tab UI and therefore closes #37

File: global.R [hunk 1]
```diff
@@ -68,7 +68,7 @@ ordered_tabs <- c(
 
 tabs <- list(
   QuestionTab$new("title", NULL, "auth", FALSE, FALSE),
-  QuestionTab$new("auth", "title", NULL),
+  QuestionTab$new("auth", "title", NULL, ui_pars = list(captcha_code=read_yaml('secret.yml')$captcha_code)),
```
Comment (JohnAdders) [inline global.R:71]: can we avoid having to read the yaml file twice?  I was expecting this to need to be done dynamically.

Comment (krzmip) [reply global.R:71]: We can avoid reading the yaml second time by reading its content once outside the server ( => in the global). Unless there are (security?) arguments not to use such global variables for secret variables

File: R/QuestionTab.R [hunk 2]
```diff
@@ -36,13 +61,20 @@ QuestionTab <- R6Class(
       self$exposure <- exposure
       self$type <- type
       self$subtype <- subtype
+      self$ui_pars <- ui_pars
```
Comment (JohnAdders) [inline R/QuestionTab.R:64]: What does pars mean?  Can we have a more useful word for this variables: settings, pairs etc

File: global.R [hunk 3]
```diff
@@ -66,9 +66,18 @@ ordered_tabs <- c(
   "report"
 )
 
+userData <- list()
+if (file.exists("secret.yml")) {
+  secret_pars <- read_yaml("secret.yml")
```
Comment (JohnAdders) [inline global.R:71]: again can we replace pars with something else.

File: global.R (hunk 1 above)
Comment (JohnAdders) [reply global.R]: thanks a lot, I'm happy with globals approach

File: R/QuestionTab.R (hunk 2 above)
Comment (krzmip) [reply R/QuestionTab.R]: renamed to "settings"

File: global.R (hunk 3 above)
Comment (krzmip) [reply global.R]: renamed to "settings"

**Pull request merged**